ideascube/management/commands/tags.py

```diff
@@ -121,3 +136,44 @@ def list(self, options):
         print(row.format('.' * 40, '.' * 40, '.' * 40))
         for tag in Tag.objects.order_by('slug'):
             print(row.format(tag.name, tag.slug, self._count(tag.name)))
+
+    def sanitize(self, options):
+        for relation in TaggedItem.objects.all():
+            tag = relation.tag
+            new_tag_name = sanitize_tag_name(tag.name)
+            if not new_tag_name:
+                # The tag was just a punctuation or a space.
+                relation.delete()
+                continue
+            # Get tags with the same case-insensitive final name.
+            tags = Tag.objects.filter(name__iexact=new_tag_name).order_by('id')
+
+            if not tags.count():
+                # There is none.
+                # This can happen if we change the tag name.
+                # Then, we just want to rename the tag.
+                tag.name = new_tag_name
+                tag.save()
+            else:
+                new_tag = tags[0]
+                if tag == new_tag:
+                    continue
```

Comment:
`new_tag` is the tag we want to keep. 
But `tag` may be already the tag we want to keep (no change in the name, the only one (or the first created) with this name). 
In this case everything is ok, just continue to the next relation.